apps/site/lib/site_web/controllers/schedule/schedule_api.ex

```diff
@@ -0,0 +1,27 @@
+defmodule SiteWeb.ScheduleController.ScheduleApi do
+  use SiteWeb, :controller
+
+  def show(conn, %{"id" => id, "date" => date, "direction_id" => direction_id}) do
+    {:ok, date} = Date.from_iso8601(date)
+    schedule_data = get_schedules(id, date, direction_id)
+
+    json(conn, schedule_data)
+  end
+
+  @spec get_schedules(binary, any, any) :: %{by_trip: map, trip_order: [any]}
+  def get_schedules(route_id, date, direction_id) do
+    services =
```

Comment:
are these `services` or `schedules`?

Reply:
Well we're getting them out of a `Schedules.Repo` so it would be weird if they weren't `schedules`. I cut-and-paste this without reading it closely, will correct.

Reply:
They are individual schedules (to comprise a full trip of schedules) organized by service.  A schedule is a "route A stops at this stop at this time", a trip is a collection of schedules (from origin -> last stop), and service is a collection of trips.

just a FYI clarification since the structure of the data from the API makes these kind of interchangeable terms, even in the frontend code.